### ISSUE TITLE
dockerng: RepoTags not iterable with docker 1.12 on ubuntu 14.04

### ISSUE THREAD

**Issue by epelc** [saltstack/salt#35139]
### Description of Issue/Question
This line causes `docker.running` to fail when using docker 1.12
https://github.com/saltstack/salt/blob/485dd07fe9b6d32cf0e9f284a71fa512d7508a6b/salt/modules/dockerng.py#L2352

Minion error log
http://pastebin.com/mKivKRLa

### Setup
(Please provide relevant configs and/or SLS files (Be sure to remove sensitive info).)
```
run:
  dockerng.running:
# the tag is important. It may not be triggered if you use the default latest. The error is in the tag lookup code.
    - image: me/my-image:my-tag
    - command: some-command
    - detach: True
    - restart_policy: always
    - environment:
      {% for key, value in pillar['config'].iteritems() %}
      - '{{ key }}': '{{ value }}'
      {% endfor %}
```

### Steps to Reproduce Issue
(Include debug logs if possible and relevant.)

Run a state using `docker.running` with an image that has a specific tag.

### Versions Report
(Provided by running `salt --versions-report`. Please also mention any differences in master/minion versions.)
```
docker version
Client:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 22:00:36 2016
 OS/Arch:      linux/amd64

Server:
 Version:      1.12.0
 API version:  1.24
 Go version:   go1.6.3
 Git commit:   8eab29e
 Built:        Thu Jul 28 22:00:36 2016
 OS/Arch:      linux/amd64

Salt Version:
           Salt: 2016.3.2
 
Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: 1.5
          gitdb: 0.5.4
      gitpython: 0.3.2 RC1
          ioflo: Not Installed
         Jinja2: 2.8
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: 0.9.1
   msgpack-pure: Not Installed
 msgpack-python: 0.4.8
   mysql-python: 1.2.3
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.6 (default, Jun 22 2015, 17:58:13)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.0.1
           RAET: Not Installed
          smmap: 0.8.2
        timelib: Not Installed
        Tornado: 4.4.1
            ZMQ: 4.0.5
 
System Versions:
           dist: Ubuntu 14.04 trusty
        machine: x86_64
        release: 3.13.0-86-generic
         system: Linux
        version: Ubuntu 14.04 trusty
```

**Comment by mikesager** [comment]
I've just run into this same issue and found this issue during my search.  I've been at it for a few minutes and I think I have info to help.  Run `docker images` and see if there are any entries with `<none>` in the `TAG` column.  That seems to be why it's choking.  I suppose there just needs to be a test for this condition, but I'm not familiar with the codebase nor am I grokking Python beyond a basic level, so this is the extent of my contribution.  As a temporary workaround, you can delete the non-tagged image with `docker rmi <image ID>`.  Hope that helps!

**Comment by Ch3LL** [comment]
@epelc can you check if you indeed are running into the same behavior as @mikesager ? do you have any images with none in the tag column. I'm not able to replicate this behavior on a clean setup using this state file:

```
run:
  dockerng.running:
    - image: ubuntu:12.04
    - command: echo "hello"
    - detach: True
    - restart_policy: always
```

So it seems that might be the key factor into why you are running into that issue.

**Comment by epelc** [comment]
@Ch3LL I can confirm having I untagged(none) images causes it. 

I partially worked around it today by removing all none images before using the docker module.

**Comment by hjc1710** [comment]
I'm quite sure adding updating [this line](https://github.com/saltstack/salt/blob/485dd07fe9b6d32cf0e9f284a71fa512d7508a6b/salt/modules/dockerng.py#L2352) to be:

```python
for repo_tag in item.get('RepoTags', []) or []:
```

Would fix the entire issue. However, I can't grok Salt's test suite and have no idea how to write a test for this to prove it =/.

**Comment by freelock** [comment]
Same issue on Ubuntu 16.04 with docker.image_present, removing all untagged images fixes...

**Comment by maguowei** [comment]
@freelock +1

**Comment by alexykot** [comment]
I've just experienced same issue. Having an untagged image causes it to fail with identical `NoneType` exception, and manually removing that image resolves it.

**Comment by Ch3LL** [comment]
@hjc1710 Our test documentation is [here](https://docs.saltstack.com/en/latest/topics/tutorials/writing_tests.html) Let me know if you have any questions still after going through that documenation if you still don't understand our test suite. Or if i can help with anything else to submit a PR. thanks

**Comment by ticosax** [comment]
has been fixed by #35447

**Comment by rallytime** [comment]
@epelc @freelock @hjc1710 See ^. Can one of you guys give that test a try and confirm this is fixed? Thanks!

**Comment by epelc** [comment]
@rallytime any way to install the specific pr easily? Can I do it with pip?

**Comment by rallytime** [comment]
@epelc You can do it that way, sure. You can also just pull down the dockerng module and put it in your `_modules` directory and sync it. Whatever way is easiest for you! :)

Here's some helpful docs on how to do that: https://docs.saltstack.com/en/latest/ref/file_server/dynamic-modules.html

**Comment by hjc1710** [comment]
@rallytime - just tested on a local Vagrant and it seemed to work wonderfully! Will commit to one of my larger projects and do a big deploy today to see if everything else works out and post here, but I think we're looking good.

For others: I placed this in `_modules/dockerng.py` in the `/srv/salt` folder. I just copied the file wholesale, and restarted the minion. It then worked.

**NOTE:** I have only tried this in a master-less configuration!
